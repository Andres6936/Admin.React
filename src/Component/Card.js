import React from 'react';

export class Card extends React.Component {
    render() {
        return <div className={"card rounded shadow-sm py-2 mb-4"}>
            {this.props.children}
        </div>
    }
}
