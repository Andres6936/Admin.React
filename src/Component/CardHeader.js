import React from 'react';

export class CardHeader extends React.Component {
    render() {
        return (
            <div className={"card-header text-primary bg-white fw-bold"}>
                {this.props.title}
                {this.props.children}
            </div>
        );
    }
}
