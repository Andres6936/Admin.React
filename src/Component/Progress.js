import React from 'react';

export class Progress extends React.Component {
    render() {
        return (
            <div className="progress mb-3">
                <div className={"progress-bar " + this.props.colorBar} role="progressbar"
                     style={{width: this.props.percentage + '%'}}
                     aria-valuenow={this.props.percentage} aria-valuemin="0" aria-valuemax="100">

                </div>
            </div>
        );
    }
}
