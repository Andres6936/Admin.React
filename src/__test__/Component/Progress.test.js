import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";
import {Progress} from "../../Component/Progress";

Enzyme.configure({adapter: new Adapter()})

describe('<Progress/>', function () {
    it("Snapshot for default Row component", () => {
        const wrapper = shallow(<Progress/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it("Snapshot for Row component", () => {
        const wrapper = shallow(<Progress colorBar={"bg-primary"} percentage={80}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})
