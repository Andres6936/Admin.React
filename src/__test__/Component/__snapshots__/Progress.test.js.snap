// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress/> Snapshot for Row component 1`] = `
<div
  className="progress mb-3"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={80}
    className="progress-bar bg-primary"
    role="progressbar"
    style={
      Object {
        "width": "80%",
      }
    }
  />
</div>
`;

exports[`<Progress/> Snapshot for default Row component 1`] = `
<div
  className="progress mb-3"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    className="progress-bar undefined"
    role="progressbar"
    style={
      Object {
        "width": "undefined%",
      }
    }
  />
</div>
`;
